app/src/main/java/com/ataulm/whatsnext/TokensStore.kt

```diff
@@ -19,6 +19,8 @@ class TokensStore private constructor(private val preferences: SharedPreferences
                 .apply()
     }
 
+    fun userIsSignedIn() = token != null
```

Comment:
This is a rudimentary check for whether the user is signed in. There's no way to tell in advance of making an API call and there's no way to guarantee that they will still be signed in after that (e.g. the token could expire immediately after we check).

This lets us head off the case where we _know_ the user isn't signed in. We'll have to deal with errors separately (e.g. network failure, missing refresh token etc)